src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs

```diff
@@ -59,6 +58,322 @@ public object ScrollOwner
 
 		private ScrollViewer Scroller => ScrollOwner as ScrollViewer;
 
+		internal object ManagedContent
+		{
+			get
+			{
+#if __ANDROID__ || __IOS__
+				if (Content is NativeScrollContentPresenter nativePresenter)
+				{
+					return nativePresenter.Content;
+				}
+#endif
+
+				return Content;
+			}
+		}
+
+		protected override void OnBringIntoViewRequested(BringIntoViewRequestedEventArgs args)
+		{
+			base.OnBringIntoViewRequested(args);
+
+			UIElement content = ManagedContent as UIElement;
+
+			if (args.Handled ||
+				args.TargetElement == this ||
+				(args.TargetElement == content && content?.Visibility == Visibility.Collapsed) ||
+				!SharedHelpers.IsAncestor(args.TargetElement, this, true /*checkVisibility*/))
+			{
+				// Ignore the request when:
+				// - There is no InteractionTracker to fulfill it.
+				// - It was handled already.
+				// - The target element is this ScrollPresenter itself. A parent scrollPresenter may fulfill the request instead then.
+				// - The target element is effectively collapsed within the ScrollPresenter.
+				return;
+			}
+
+			Rect targetRect = new Rect();
+			//int offsetsChangeCorrelationId = 0;
+			double targetZoomedHorizontalOffset = 0.0;
+			double targetZoomedVerticalOffset = 0.0;
+			double appliedOffsetX = 0.0;
+			double appliedOffsetY = 0.0;
+			var viewportWidth = ViewportWidth;
+			var viewportHeight = ViewportHeight;
+			var zoomFactor = Scroller.ZoomFactor;				
+
+			// Compute the target offsets based on the provided BringIntoViewRequestedEventArgs.
+			ComputeBringIntoViewTargetOffsets(
+				content,
+				args,
+				out targetZoomedHorizontalOffset,
+				out targetZoomedVerticalOffset,
+				out appliedOffsetX,
+				out appliedOffsetY,
+				out targetRect);
+
+			// Do not include the applied offsets so that potential parent bring-into-view contributors ignore that shift.
+			Rect nextTargetRect = new Rect(
+				targetRect.X * zoomFactor - targetZoomedHorizontalOffset - appliedOffsetX,
+				targetRect.Y * zoomFactor - targetZoomedVerticalOffset - appliedOffsetY,
+				Math.Min(targetRect.Width * zoomFactor, viewportWidth),
+				Math.Min(targetRect.Height * zoomFactor, viewportHeight));
+
+			Rect viewportRect = new Rect(
+				0.0f,
+				0.0f,
+				(float)viewportWidth,
+				(float)viewportHeight);
+
+			var verticalOffset = Scroller.VerticalOffset;
+			var horizontalOffset = Scroller.HorizontalOffset;
+			var zoomedVerticalOffset = verticalOffset;
+			var zoomedHorizontalOffset = horizontalOffset;
+
+			if (targetZoomedHorizontalOffset != zoomedHorizontalOffset ||
+				targetZoomedVerticalOffset != zoomedVerticalOffset)
+			{
+				//com_ptr<ScrollingScrollOptions> options =
+				//	make_self<ScrollingScrollOptions>(
+				//		args.AnimationDesired() ? ScrollingAnimationMode::Auto : ScrollingAnimationMode::Disabled,
+				//		snapPointsMode);
```

Comment:
Should we move that method to a `SCP.mux.cs` file, or it's only "inspired" by the MUX version?

Reply:
Ok then it's "inspired" by, so no needs to create a `SCP.mux.cs` file. Wondering however if we should keep that commented code or not ... your choice ! :)